source/includes/code-examples/DeleteOneAsync.cs

```diff
@@ -0,0 +1,75 @@
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization.Attributes;
+using MongoDB.Bson.Serialization.Conventions;
+using MongoDB.Driver;
+using static System.Console;
+
+namespace CSharpExamples.UsageExamples;
+
+public class DeleteOneAsync
+{
+    private static IMongoCollection<Restaurant> _restaurantsCollection;
+    private static string _mongoConnectionString = "<Your MongoDB URI>";
+
+    public static void Main(string[] args)
+    {
+        Setup();
+
+        var doc = _restaurantsCollection.Find(Builders<Restaurant>.Filter
+            .Eq(r => r.Name, "Ready Penny Inn")).First();
+
+        // Delete a document using builders
+        WriteLine("Deleting a document with builders...");
+        var result = DeleteARestaurantBuilderAsync();
+
+        WriteLine($"Deleted documents: {result.Result.DeletedCount}");
```

Comment:
Except for a few cases, it's a bad practice to wait for task with `Task.Result`.
I'd suggest to make Main async:   `async Task Main(string[] args)` and 
`var deleteRersult = await DeleteARestaurantBuilderAsync();` in all async examples.

Reply:
Agreed. Main should be async